### PR TITLE
Support branching for refresh chromebot status

### PR DESCRIPTION
**Pull request by keyonghan**
This is part of https://github.com/flutter/flutter/issues/51807

### CODE REVIEW

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart [hunk 1]
```diff
@@ -7,22 +7,47 @@ import 'package:cocoon_service/src/model/appengine/task.dart';
 import 'package:cocoon_service/src/request_handlers/refresh_chromebot_status.dart';
```
Comment (CaseyHillers) [inline app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: nitty nit: these package:cocoon imports should be between the import package blocks and the relative imports for the test files block

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart [hunk 2]
```diff
@@ -63,12 +104,65 @@ void main() {
...
+      expect(taskMaster.status, Task.statusSucceeded);
+      expect(taskOther.status, Task.statusSucceeded);
     });
   });
 }
 
 // ignore: must_be_immutable
 class MockLuciService extends Mock implements LuciService {}
+
+class MockGitHub extends Mock implements GitHub {}
```
Comment (CaseyHillers) [inline app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: These mocks could also be moved to the test utils file to remove the need for creating them in each file

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart [hunk 3]
```diff
@@ -7,22 +7,47 @@ import 'package:cocoon_service/src/model/appengine/task.dart';
...
 void main() {
   group('RefreshChromebotStatus', () {
     FakeConfig config;
     ApiRequestHandlerTester tester;
     MockLuciService mockLuciService;
     RefreshChromebotStatus handler;
+    List<String> githubBranches;
+    FakeHttpClient branchHttpClient;
+
+    Stream<Branch> branchStream() async* {
```
Comment (CaseyHillers) [inline app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Branch stream seems to be used through the tests. We should move it to a test utils file to remove the duplication through the codebase.

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Reorganized 😄

File: app_dart/lib/src/request_handlers/refresh_chromebot_status.dart [hunk 4]
```diff
@@ -49,30 +59,48 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
...
+                transaction.queueMutations(inserts: <Task>[update]);
+                // Stop updating task whenever we find the latest status of the same commit.
+                break;
+              }
+            }
+          }
+        }
+        await transaction.commit();
+      } catch (error) {
+        rethrow;
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/refresh_chromebot_status.dart]: This is some existing logic there without further changes. I just changed to catch/log the error info.

File: app_dart/lib/src/request_handlers/refresh_chromebot_status.dart [hunk 5]
```diff
@@ -49,30 +59,48 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
...
+          datastore,
+          luciTasks,
+        );
       });
     }
-
     return Body.empty;
   }
+
+  Future<void> _updateStatus(
```
[the remark this answers is not in the document]
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/refresh_chromebot_status.dart]: Added.

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart (hunk 3 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: This is a good point. At least five APIs are using this code. Created a TODO to simply test codes, and prefer to work on it in a separate PR.

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Ditto.

File: app_dart/lib/src/request_handlers/refresh_chromebot_status.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/refresh_chromebot_status.dart]: RequestHandler already does this https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handling/request_handler.dart#L42-L62

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart (hunk 3 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Tracking issue: https://github.com/flutter/flutter/issues/53280

File: app_dart/lib/src/request_handlers/refresh_chromebot_status.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/refresh_chromebot_status.dart]: Ha, removed the try catch!

**Pull request merged**